dedupe/dedupe.go

```diff
@@ -0,0 +1,59 @@
+package dedupe
+
+// MaxInMemoryDedupeSize (default : 100 MB)
+var MaxInMemoryDedupeSize = 100 * 1024 * 1024
+
+type DedupeBackend interface {
+	// Upsert add/update key to backend/database
+	Upsert(elem string)
+	// Execute given callback on each element while iterating
+	IterCallback(callback func(elem string))
+	// Cleanup cleans any residuals after deduping
+	Cleanup()
+}
+
+// Dedupe is string deduplication type which removes
+// all duplicates if
+type Dedupe struct {
+	receive <-chan string
+	backend DedupeBackend
+}
+
+// Drains channel and tries to dedupe it
+func (d *Dedupe) Drain() {
+	for {
+		val, ok := <-d.receive
+		if !ok {
+			break
+		}
+		d.backend.Upsert(val)
+	}
+}
+
```

Comment:
@dogancanbakir in tlsx we are printing unique results after end of program in cli mode (i.e not-json). as @ehsandeep was also suggesting . we can implement something similar to [anew](https://github.com/tomnomnom/anew) i.e we can modify `upsert`  to `Upsert(elem string) bool` and it will return true if elem is duplicated else false and if false we can pipe it to other channel and this will give a stream of unique results 

What do you think ?

Reply:
@tarunKoyalwar,
I made some changes. Could you take another look and give me your thoughts?